beluga/include/beluga/storage.hpp

```diff
@@ -179,10 +179,11 @@ class StoragePolicy : public Mixin {
   void initialize_particles(Range&& input) {
     static_assert(std::is_same_v<particle_type, ranges::range_value_t<Range>>, "Invalid value type");
     const std::size_t size = this->self().max_samples();
-    particles_.resize(size);
-    const auto first = std::begin(views::all(particles_));
+    particles_[1].resize(size);
```

Comment:
Mutating while iterating issues never triggered SEGFAULTs because `particles_` is always expanded to fit `max_samples` and [`std::vector::resize`](https://en.cppreference.com/w/cpp/container/vector/resize) does not change vector capacity.